src/overlays/actors/ovl_En_Weather_Tag/z_en_weather_tag.c

```diff
@@ -26,45 +42,296 @@ const ActorInit En_Weather_Tag_InitVars = {
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Weather_Tag/func_80B318C0.s")
+void func_80B31AA0(EnWeatherTag* this, GlobalContext* globalCtx) {
+    if (func_80B3149C(this, globalCtx, 1, 0, 4, 0, 0x64) != 0) {
+        func_80077684(globalCtx);
+        globalCtx->envCtx.lightning = 2;
+        globalCtx->envCtx.unk_EE[0] = 0;
+        func_80B31020(this, func_80B31A10);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Weather_Tag/func_80B31924.s")
+void func_80B31B24(EnWeatherTag* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Weather_Tag/func_80B319A0.s")
+    if (func_8002DB8C(&player->actor, &this->actor) < ((this->actor.params >> 8) * 100.0f)) {
+        Math_SmoothScaleMaxMinS(&globalCtx->envCtx.unk_9E, -0x50, 1, 2, 1);
+        Math_SmoothScaleMaxMinS(&globalCtx->envCtx.unk_A0, -0x7D0, 1, 0x32, 1);
+        return;
+    }
+    Math_SmoothScaleMaxMinS(&globalCtx->envCtx.unk_9E, 0, 1, 1, 1);
+    Math_SmoothScaleMaxMinS(&globalCtx->envCtx.unk_A0, 0, 1, 0x19, 1);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Weather_Tag/func_80B31A10.s")
+void func_80B31C20(EnWeatherTag* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Weather_Tag/func_80B31AA0.s")
+    if (func_8002DB8C(&player->actor, &this->actor) < ((this->actor.params >> 8) * 100.0f)) {
+        func_80077624(globalCtx);
+        globalCtx->envCtx.lightning = 1;
+        globalCtx->envCtx.unk_EE[0] = 0x19;
+        func_80B31020(this, func_80B31CC0);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Weather_Tag/func_80B31B24.s")
+void func_80B31CC0(EnWeatherTag* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Weather_Tag/func_80B31C20.s")
+    if ((((this->actor.params >> 8) * 100.0f) + 10.0f) < func_8002DB8C(&player->actor, &this->actor)) {
+        func_80077684(globalCtx);
+        globalCtx->envCtx.lightning = 2;
+        globalCtx->envCtx.unk_EE[0] = 0;
+        globalCtx->envCtx.unk_EE[1] = 0xA;
+        func_80B31020(this, func_80B31C20);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Weather_Tag/func_80B31CC0.s")
+void EnWeatherTag_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnWeatherTag* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Weather_Tag/EnWeatherTag_Update.s")
+    this->actionfunc(this, globalCtx);
+    if (BASE_REG(0x19, 0) != 0) {
+        DebugDisplay_AddObject(this->actor.posRot.pos.x, this->actor.posRot.pos.y, this->actor.posRot.pos.z,
+                               this->actor.posRot.rot.x, this->actor.posRot.rot.y, this->actor.posRot.rot.z, 1.0f, 1.0f,
+                               1.0f, 0xFF, 0, 0xFF, 0xFF, 4, globalCtx->state.gfxCtx);
```

Comment:
These are color values so they should be decimal as of recently
```suggestion
                               1.0f, 255, 0, 255, 255, 4, globalCtx->state.gfxCtx);
```